Plugin incompatible with configuration cache projects

### Describe the bug

When trying to use the plugin (either by itself or through the VSCode/IntelliJ JFrog plugin) in a Gradle project with the configuration cache enabled, it fails with an exception.

### Current behavior

When using the IntelliJ plugin to initiate an XRay scan, I get the following error:

```
Xray scan failed: IOException: 

FAILURE: Build failed with an exception.

* What went wrong: Configuration cache problems found in this build. 3 problems were found storing the configuration cache, 1 of which seems unique.
- Task `:myModule:generateDepTrees` of type `com.jfrog.tasks.GenerateDepTrees`: invocation of 'Task.project' at execution time is unsupported. See https://docs.gradle.org/8.14/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```

### Reproduction steps

Run the `generateDepTrees` task in a Gradle project with `org.gradle.configuration-cache = true` in its gradle.properties file

OR

Use the IntelliJ plugin to initiate a local XRay scan in such  a Gradle project with `org.gradle.configuration-cache = true` in its gradle.properties file.

### Expected behavior



### Gradle Dependency Tree version

3.0.2

### Operating system type and version

Windows 10

--- a/src/functionalTest/java/com/jfrog/tasks/FunctionalTestBase.java
+++ b/src/functionalTest/java/com/jfrog/tasks/FunctionalTestBase.java
@@ -27,6 +27,6 @@ public class FunctionalTestBase {
 
     @DataProvider
     public Object[][] gradleVersions() {
-        return new Object[][]{{"5.6.4"}, {"6.9"}, {"7.4.2"}, {"7.6"}};
+        return new Object[][]{{"5.6.4"}, {"6.9"}, {"7.4.2"}, {"7.6"}, {"8.14.2"}};
     }
 }

--- a/src/functionalTest/java/com/jfrog/tasks/IncludeAllBuildFilesTest.java
+++ b/src/functionalTest/java/com/jfrog/tasks/IncludeAllBuildFilesTest.java
@@ -24,6 +24,8 @@ import static org.testng.Assert.fail;
  * Functional tests for the project under resources/multi/
  * This project contain subprojects in a multiple build.gradle files.
  * This project tests the INCLUDE_ALL_BUILD_FILES flag, which should add all the subprojects under resources/multi/.
+ * It uses the configuration cache on Gradle 8.1+.
+ * 
  * @author omerz
  **/
 public class IncludeAllBuildFilesTest extends FunctionalTestBase {

--- a/src/functionalTest/java/com/jfrog/tasks/MultiProjectsTest.java
+++ b/src/functionalTest/java/com/jfrog/tasks/MultiProjectsTest.java
@@ -22,6 +22,7 @@ import static org.testng.Assert.*;
 /**
  * Functional tests for the project under resources/multi/
  * This project contain subprojects in a multiple build.gradle files.
+ * It uses the configuration cache on Gradle 8.1+.
  *
  * @author yahavi
  **/

--- a/src/functionalTest/resources/multi/gradle.properties
+++ b/src/functionalTest/resources/multi/gradle.properties
@@ -1,1 +1,3 @@
 currentVersion=1.0-SNAPSHOT
+
+org.gradle.configuration-cache=true

--- a/src/main/java/com/jfrog/tasks/GenerateDepTrees.java
+++ b/src/main/java/com/jfrog/tasks/GenerateDepTrees.java
@@ -18,6 +18,7 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.*;
+import java.util.stream.Collectors;
 
 import static com.jfrog.GradleDependencyTreeUtils.addConfiguration;
 
@@ -43,6 +44,14 @@ public class GenerateDepTrees extends DefaultTask {
             }
             return true;
         });
+
+        // On Gradle 7.4+, mark this task as incompatible with the configuration cache
+        List<Integer> gradleVersionParts = Arrays.stream(getProject().getGradle().getGradleVersion().split("\\."))
+                .map(Integer::valueOf)
+                .collect(Collectors.toList());
+        if (gradleVersionParts.get(0) >= 8 || (gradleVersionParts.get(0) == 7 && gradleVersionParts.get(1) >= 4)) {
+            notCompatibleWithConfigurationCache("Accesses projects at execution time");
+        }
     }
 
     @Internal

